sdks/java/io/google-cloud-platform/build.gradle

```diff
@@ -76,3 +76,15 @@ dependencies {
   testCompile library.java.junit
   testCompile library.java.slf4j_jdk14
 }
+
+// Shade dependencies.
+shadowJar {
+  dependencies {
+    include(dependency(library.java.grpc_netty))
+    include(dependency(library.java.netty_handler))
+    include(dependency(library.java.netty_tcnative_boringssl_static))
```

Comment:
tcnative is a package containing JNI code, just relocating it doesn't always work since the JNI bindings and java interfaces need to lineup.

See:
https://stackoverflow.com/questions/46650776/after-shading-io-netty-tcnative-stops-working
https://github.com/netty/netty/issues/7272

Were you able to test that this shading worked?
Also, this can be very brittle, can you add a test to show that shading of tcnative is working?